src/main/java/com/joejoe2/chat/service/channel/GroupChannelServiceImpl.java

```diff
@@ -47,7 +47,7 @@
 @Service
 public class GroupChannelServiceImpl implements GroupChannelService {
   private static final Logger logger = LoggerFactory.getLogger(GroupChannelService.class);
-  @Autowired UserRepository userRepository;
+  @Autowired UserService userService;
```

Comment:
In general, it's recommended to have a constructor injection instead due to immutable objects, increased testability, and SRP.

Is it planned in the near future to make it like this?

Fields themselves need to be private final

See - https://www.baeldung.com/java-spring-field-injection-cons